### ISSUE TITLE
`Rails/HasManyOrHasOneDependent` ignores nested `with_options`

### ISSUE THREAD

**Issue by yskkin** [bbatsov/rubocop#5784]
## Expected behavior

No offences are detected for `has_many :special_tags`.

## Actual behavior

`Rail/HasManyOrHasOneDependent` is detected for `has_many :special_tags`

## Steps to reproduce the problem

Lint following model:

```
class Article < ApplicationRecord
  with_options dependent: :destroy do
    has_many :tags
    with_options class_name: "Tag" do
      has_many :special_tags, foreign_key: :special_id, inverse_of: :special
    end
  end
end
```

## RuboCop version

```
rubocop -V
0.54.0 (using Parser 2.5.0.5, running on ruby 2.5.0 x86_64-darwin17)
```
